packages/core/src/util/LowPrecisionTaskScheduler/LowPrecisionTaskScheduler.ts

```diff
@@ -52,11 +52,11 @@ const addTask = ({key, firingDate, task, intervalDelay}: ScheduleLowPrecisionTas
     const tasks = intervals[intervalDelay]?.tasks;
     if (tasks?.length !== 0) {
       for (const taskData of tasks) {
-        if (nowTime >= firingDate) {
+        if (nowTime >= taskData.firingDate) {
```

Comment:
We should compare with `firingDate` of each task separately, not with with `firingDate` that is passed via props when adding a new task (it would be the same for each task we check in the loop)